iOS/IssueTracker/IssueTracker/LogIn/View/SignUp/SignUpViewController.swift

```diff
@@ -15,26 +16,146 @@ class SignUpViewController: UIViewController {
+    func idErrorLabelUpdate(id: String) {
+        if id.isEmpty {
+            self.idErrorMessageLabel.isHidden = true
+        } else {
+            self.idErrorMessageLabel.text = id
+            self.idErrorMessageLabel.isHidden = false
+        }
+    }
+    
+    func passwordErrorLabelUpdate(password: String) {
+        if password.isEmpty {
+            self.passwordErrorMessageLabel.isHidden = true
+        } else {
+            self.passwordErrorMessageLabel.text = password
+            self.passwordErrorMessageLabel.isHidden = false
+        }
+    }
+        
+    func showToast(message : String) {
+        let width_variable:CGFloat = 100
+        let toastLabel = UILabel(frame: CGRect(x: width_variable, y: self.view.frame.size.height-100, width: view.frame.size.width-2*width_variable, height: 35))
+        toastLabel.backgroundColor = UIColor.black.withAlphaComponent(0.6)
+        toastLabel.textColor = UIColor.white
+        toastLabel.textAlignment = .center;
+        toastLabel.font = UIFont(name: "Montserrat-Light", size: 12.0)
+        toastLabel.text = message
+        toastLabel.alpha = 1.0
+        toastLabel.layer.cornerRadius = 10;
+        toastLabel.clipsToBounds  =  true
+        
+        UIApplication.shared.keyWindow?.addSubview(toastLabel)
+        
+        UIView.animate(withDuration: 4.0, delay: 0.1, options: .curveEaseOut, animations: {
+            toastLabel.alpha = 0.0
+        }, completion: {(isCompleted) in
+            toastLabel.removeFromSuperview()
+        })
+    }
+    
+    @IBAction func touchedSignUp(_ sender: Any) {
+        
+        let url = "http://172.30.1.27:5000"
+        
+        let parameters = ["userId": self.idTextField.text,
+                          "password": self.passwordTextField.text,
+                          "nickname": self.nickNameTextField.text]
+        
+        let headers: HTTPHeaders = ["Accept": "application/json"]
+        
+        AF.request(url + "/api/signup", method: .post, parameters: parameters, headers: headers).responseJSON { (response) in
+            switch response.result {
+            case let .success(json):
+                print(json)
+                if let dic = json as? [String: String] {
+                    let message = dic["message"] ?? ""
+                    print(message)
+                }
+            case let .failure(error):
+                print(error)
+            }
+            self.navigationController?.popViewController(animated: true)
+        }
+    }
+}
 
 
-
-
+extension SignUpViewController: UITextFieldDelegate {
+    func textFieldDidChangeSelection(_ textField: UITextField) {
+        viewModel.action.idTextFieldChanged(self.idTextField.text!)
+        viewModel.action.passwordTextFieldChanged(self.passwordTextField.text!)
+        viewModel.action.passwordConfirmFieldChanged(self.passwordTextField.text!, self.passwordConfirmTextField.text!)
+        viewModel.action.nicknameFieldChanged(self.nickNameTextField.text!)
```

Comment:
[textField 강제 언래핑 vs textLabel 강제 언래핑](https://stackoverflow.com/questions/53572568/force-unwrapping-uitextfield-text-not-crash-but-uilabel-text-crash) 
여기서 강제 언래핑해도 괜찮은 건지 궁금하서 한 번 검색해 봤는데요, 

If a label has no text, then label.text will return nil. However, if a text field has no text then textField will return "" (an empty string). That's why it will not crash if you force-unwrap the text property of a text field.

라고 합니다. 아직 정확한 이유는 모르겠지만, UILabel의 text를 사용할 떄는 조심해야겠네ㅎㅎ